explainer.py

```diff
@@ -0,0 +1,164 @@
+import os
+import time
+import json
+import asyncio
+from datetime import datetime
+from presentation_parser import PresentationParser
+from gpt_slide_explainer import GPTSlideExpander
+from db.db_engine import Upload, UploadStatus, get_engine
+from sqlalchemy.orm import Session
+
+UPLOADS_FOLDER = 'uploads'
+OUTPUTS_FOLDER = 'outputs'
+STATUS_FILE = 'status.json'
+SLEEP_TIME = 10  # seconds
+
+
+def process_files():
+    """
+    Continuously processes unprocessed files.
+
+    The function scans the uploads folder for unprocessed files and processes each file one by one.
+    After processing, it saves the explanation, updates the status, and sleeps for a specified duration.
+    """
+    while True:
+        engine = get_engine()
+        with Session(engine) as session:
+            pending_status = UploadStatus.pending
+            upload_files = session.query(Upload).filter_by(status=pending_status).all()
+            for upload_file in upload_files:
+                try:
+                    upload_path = f"{UPLOADS_FOLDER}/{upload_file.filename}"
+                    presentation = load_presentation(upload_path)
+                    explanation = generate_explanation(presentation)
+                    upload_file.finish_time = datetime.now().strftime('%Y-%m-%d-%H-%M-%S')
+                    upload_file.status = UploadStatus.done
+                    output_file = save_explanation(explanation, upload_path)
+                    print(f"Saved explanation to: {output_file}")
+                    session.commit()
+                except KeyError:
+                    continue
+        # unprocessed_files = find_unprocessed_files()
+        #
+        # for file_path in unprocessed_files:
+        #     print(f"Processing file: {file_path}")
+        #
+        #     presentation = load_presentation(file_path)
+        #     explanation = generate_explanation(presentation)
+        #
+        #     output_file = save_explanation(explanation, file_path)
+        #     print(f"Saved explanation to: {output_file}")
+        #
+        #     update_status(file_path)
+
+        time.sleep(SLEEP_TIME)
+
+
+def find_unprocessed_files():
+    """
+    Finds unprocessed files in the uploads folder.
+
+    Returns:
+        List[str]: List of file paths of unprocessed files.
+    """
+    status = load_status()
+    processed_files = set(status.keys())
+    unprocessed_files = []
+
+    for file_name in os.listdir(UPLOADS_FOLDER):
+        if file_name not in processed_files:
+            unprocessed_files.append(os.path.join(UPLOADS_FOLDER, file_name))
+
+    return unprocessed_files
+
+
+def load_status():
+    """
+    Loads the status file.
+
+    Returns:
+        dict: Dictionary containing the status information.
+    """
+    if not os.path.exists(STATUS_FILE):
+        return {}
+
+    with open(STATUS_FILE) as f:
```

Comment:
This name does not adhere to python's naming conventions. Names should be indicative, and when used properly can reduce redundant documentation. In addition, each Variable, function, module, constant and class names has a specific naming convention. Read this link to learn the proper conventions - https://visualgit.readthedocs.io/en/latest/pages/naming_convention.html. If you don't know what CamelCase and snake_case means, read this link - https://www.google.com/amp/s/www.theserverside.com/answer/Camel-case-vs-snake-case-Whats-the-difference%3famp=1 . For a general overview on python conventions (PEP-8), read this link - https://peps.python.org/pep-0008